test/jdk/java/awt/print/PrinterJob/PrintLatinCJKTest.java

```diff
@@ -23,55 +23,71 @@
 import java.awt.print.PrinterJob;
+import java.lang.reflect.InvocationTargetException;
 import javax.swing.JButton;
 import javax.swing.JFrame;
-import javax.swing.JTextArea;
 
-import javax.swing.SwingUtilities;
+import static javax.swing.SwingUtilities.invokeAndWait;
 
-public class PrintLatinCJKTest implements Printable, ActionListener {
+public class PrintLatinCJKTest implements Printable {
 
-    static PrintLatinCJKTest testInstance = new PrintLatinCJKTest();
-    private PageFormat pf;
+    private static PrintLatinCJKTest testInstance = new PrintLatinCJKTest();
+    private static JFrame frame;
+    private static String info = """
+            To test 8022536, if a remote printer is the system default,
+            it should show in the dialog as the selected printer.
+            You need a printer for this test. If you have none, let
+            the test pass. If there is a printer, press Print, send
+            the output to the printer, and examine it. It should have
+            text looking like this : \u4e00\u4e01\u4e02\u4e03\u4e04English
+            """;
 
-    static String info =
-       "To test 8022536, if a remote printer is the system default,"+
-       "it should show in the dialog as the selected printer.\n"+
-       "You need a printer for this test. If you have none, let "+
-       "the test pass. If there is a printer, press Print, send "+
-       "the output to the printer, and examine it. It should have "+
-       "text looking like this : \u4e00\u4e01\u4e02\u4e03\u4e04English.";
+    public static void showFrame() throws InterruptedException, InvocationTargetException {
+        invokeAndWait( () -> {
+            frame = new JFrame("Test Frame");
+            JButton b = new JButton("Print");
+            b.addActionListener((ae) -> {
+                try {
+                    PrinterJob job = PrinterJob.getPrinterJob();
+                    job.setPrintable(testInstance);
+                    if (job.printDialog()) {
+                        job.print();
+                    }
+                } catch (PrinterException ex) {
+                    ex.printStackTrace();
+                }
+            });
+            frame.add("South", b);
+            frame.pack();
+            frame.setLocationRelativeTo(null);
+            frame.setVisible(true);
+        });
+    }
 
-    public static void showFrame() {
-         JFrame f = new JFrame();
-         JTextArea jta = new JTextArea(info, 4, 30);
-         jta.setLineWrap(true);
-         jta.setWrapStyleWord(true);
-         f.add("Center", jta);
-         JButton b = new JButton("Print");
-         b.addActionListener(testInstance);
-         f.add("South", b);
-         f.pack();
-         f.setVisible(true);
+    private static void disposeTestFrame() throws InterruptedException, InvocationTargetException {
+        invokeAndWait(() -> {
+            if (frame != null) {
+                frame.dispose();
+            }
+        });
```

Comment:
I just thought whether it's feasible to keep a list of frames / windows to be disposed of inside the `PassFailJFrame` class. It disposes of its instruction frame when the wait is over, it can handle disposing of other frames too.

If I remember correctly, the first version disposed of of all windows. If we make it an explicit list, it shouldn't cause issues.

Reply:
Fixed